src/compas_fab/backends/pybullet/client.py

```diff
@@ -175,44 +184,103 @@ def load_robot(self, urdf_file):
         """
         robot_model = RobotModel.from_urdf_file(urdf_file)
         robot = Robot(robot_model, client=self)
+        robot.attributes['pybullet'] = {}
+        if resource_loaders:
+            robot_model.load_geometry(*resource_loaders)
+            self.cache_robot(robot)
+        else:
+            robot.attributes['pybullet']['cached_robot'] = robot.model
```

Comment:
Is there a danger of recursively creating model inside model when serializing this?

Reply:
AFAIK, there is no serialization for Robots, only the things a Robot is composed of such as RobotModels.  So, no.